### ISSUE TITLE
CXX C11 Atomic support

### ISSUE THREAD

**Issue by PetervdPerk-NXP** [apache/nuttx#11547]
If you've got a C C11 library that's using `<stdatomic.h>` and you want to use this library in CXX you simply can't. C++ provides a compatibility header `<atomic>` but if you use the NuttX libcxxmini library it doesn't provide a atomic implementation. Unless you use of-course the std libraries but these tend to big.

It would interesting if we could have a minimal `<atomic>`implementation for libcxxmini.

Do note it seems that C++23 partially fixes this incompatibility.
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p0943r6.html

Would be interesting if it's possible to backport that.

**Comment by xiaoxiang781216** [comment]
Yes, @hujun260 hit the similar issue recently. atomic is a good addition to libcxxmini.
